pystac_client/warnings.py

```diff
@@ -0,0 +1,81 @@
+from contextlib import contextmanager
+from typing import Iterator, Union
+import warnings
+
+from pystac_client.conformance import ConformanceClasses
+
+
+FALLBACK_MSG = "Falling back to pystac. This might be slow."
+
+
+def DOES_NOT_CONFORM_TO(conformance_class: Union[str, ConformanceClasses]) -> str:
```

Comment:
```suggestion
def does_not_conform_to(conformance_class: Union[str, ConformanceClasses]) -> str:
```

I haven't seen this pattern before, do you have any prior art that you could point me to? I'm not sure this use-case is worth breaking function naming conventions.

Reply:
No in an old job we did this pattern exclusively for messages, but I agree. What I think would be nice would be default messages on the Warning classes themselves.

Reply:
I think another comment talks about this w.r.t. adding a `warn` classmethod to the warnings themselves to do All The Warning Stuff™.

Reply:
ooh yeah the verbosity of warnings was bumming me out.